caas/kubernetes/provider/bootstrap.go

```diff
@@ -121,9 +121,9 @@ type controllerStacker interface {
 	Deploy() error
 }
 
-func controllerCorelation(broker *kubernetesClient) (*kubernetesClient, error) {
+func controllerCorelation(broker *kubernetesClient) (string, error) {
 	if broker.Config().Name() != environsbootstrap.ControllerModelName {
```

Comment:
I think this check should go into the Open() method.
The Only reason we can to call controllerCorelation() is when we are trying to find the namespace used for the controller model. Otherwise the namespace to use is just the model name. And with this refactor, we should adjust the logic. The logic the caller uses to check ns == "" etc is a bit convoluted and things can be simlified.

Reply:
Take a look at that change.

Reply:
With the refactoring below, we no longer need
```
if broker.Config().Name() != environsbootstrap.ControllerModelName {
}
```
here

Reply:
Done & Done